crates/circuit/src/imports.rs

```diff
@@ -255,42 +255,65 @@ static STDGATE_IMPORT_PATHS: [[&str; 2]; STANDARD_GATE_SIZE] = [
 ///
 /// NOTE: the order here is significant it must match the StandardGate variant's number must match
 /// index of it's entry in this table. This is all done statically for performance
-static mut STDGATE_PYTHON_GATES: GILOnceCell<[Option<PyObject>; STANDARD_GATE_SIZE]> =
-    GILOnceCell::new();
-
-#[inline]
-pub fn populate_std_gate_map(py: Python, rs_gate: StandardGate, py_gate: PyObject) {
-    let gate_map = unsafe {
-        match STDGATE_PYTHON_GATES.get_mut() {
-            Some(gate_map) => gate_map,
-            None => {
-                let array: [Option<PyObject>; STANDARD_GATE_SIZE] = std::array::from_fn(|_| None);
-                STDGATE_PYTHON_GATES.set(py, array).unwrap();
-                STDGATE_PYTHON_GATES.get_mut().unwrap()
-            }
-        }
-    };
-    let gate_cls = &gate_map[rs_gate as usize];
-    if gate_cls.is_none() {
-        gate_map[rs_gate as usize] = Some(py_gate.clone_ref(py));
-    }
-}
+static STDGATE_PYTHON_GATES: [GILOnceCell<PyObject>; STANDARD_GATE_SIZE] = [
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+    GILOnceCell::new(),
+];
```

Comment:
This is pretty ugly. But I don't think there is a syntax that we can use here with GilOnceCell given the constraints defined in: https://doc.rust-lang.org/std/primitive.array.html so this is fine.

Reply:
Yeah, it's ugly.  For anybody from the future looking: the trouble is that there's not enough available in `const` functions to manage this.  The closest I managed to get was
```rust
use std::mem::MaybeUninit;
use pyo3::prelude::*;
use pyo3::sync::GILOnceCell;

const STANDARD_GATE_SIZE: usize = 52;

static STDGATE_PYTHON_GATES: [GILOnceCell<Py<PyAny>>; STANDARD_GATE_SIZE] = const {
    let mut out: [MaybeUninit<GILOnceCell<Py<PyAny>>>; STANDARD_GATE_SIZE] =
        [const { MaybeUninit::uninit() }; STANDARD_GATE_SIZE];
    let mut i = 0;
    while i < STANDARD_GATE_SIZE {
        out[i].write(GILOnceCell::new());  // error: `MaybeUninit::write` isn't const stable
        i += 1;
    }
    unsafe { ::std::mem::transmute(out) }
};
```